File: framework/src/main/java/com/github/mrmitew/skeleton/di/fragment/FragmentModule.java
package com.github.mrmitew.skeleton.di.fragment;

import android.support.v4.app.Fragment;

import dagger.Module;
import dagger.Provides;

@Module
public abstract class FragmentModule<T extends Fragment> {
    protected final T fragment;

    public FragmentModule(T fragment) {
        this.fragment = fragment;
    }

    @Provides
    @FragmentScope
    public T provideFragment() {
        return fragment;
    }
}

File: framework/src/main/java/com/github/mrmitew/skeleton/di/fragment/FragmentModule.kt
package com.github.mrmitew.skeleton.di.fragment

import android.app.Fragment
import dagger.Module
import dagger.Provides

@Module
abstract class FragmentModule<out T: Fragment> (protected val fragment:T) {

    @Provides
    @FragmentScope
    fun provideFragment() : T {
        return fragment
    }
}